### PR TITLE
capi: Add imported globals to instantiate

### PR DESCRIPTION
**Pull request by gumb0**

### CODE REVIEW

File: include/fizzy/fizzy.h [hunk 1]
```diff
@@ -148,14 +173,23 @@ FizzyInstance* fizzy_instantiate(const FizzyModule* module,
...
+///                                      imported_globals_size equals 0.
+/// @param      imported_globals_size    Size of the imported global array. Can be zero.
 /// @returns    non-NULL pointer to instance in case of success, NULL otherwise.
 ///
 /// @note
 /// Functions in @a imported_functions are allowed to be in any order and allowed to include some
 /// functions not required by the module.
 /// Functions are matched to module's imports based on their module and name strings.
+/// @note
+/// Function expects @a imported_globals to be in the order of imports defined in the module.
```
Comment (gumb0) [inline include/fizzy/fizzy.h:186]: It's weird how resolve_instantiate currently doesn't resolve globals the way it resolves functions. C++ backend doesn't allow this currently, and I don't know how important it is.

File: include/fizzy/fizzy.h [hunk 2]
```diff
@@ -20,12 +20,12 @@ typedef struct FizzyInstance FizzyInstance;
 /// The data type representing numeric values.
 ///
 /// i64 member is used to represent values of both i32 and i64 type.
-union FizzyValue
+typedef union FizzyValue
```
Comment (axic) [inline include/fizzy/fizzy.h:23]: Hmm, wonder how the Rust binding will act with this, will check.

File: include/fizzy/fizzy.h (hunk 1 above)
Comment (axic) [reply include/fizzy/fizzy.h:186]: I agree there should be a function similar to functions to resolve globals in the C++ API. That could be used here.

Probably at that point I'd rename the to instantiates as `fizzy_instantiate` (which does resolution) and `fizzy_unsafe_instantiate` (which doesn't).

**Review by axic — APPROVED** [review]
Haven't done a very thorough review of the last two commits, but in general it looks good to me.

File: include/fizzy/fizzy.h (hunk 2 above)
Comment (axic) [reply include/fizzy/fizzy.h:23]: Seems to be working well.

**Review by chfast — APPROVED** [review]

**Pull request merged**